Headers/DebugServer2/Host/File.h

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Host_File_h
+#define __DebugServer2_Host_File_h
+
+#include "DebugServer2/Base.h"
+#include "DebugServer2/Constants.h"
+#include "DebugServer2/Utils/Log.h"
+
+#include <string>
+
+namespace ds2 {
+namespace Host {
+
+class File {
+public:
+  File(std::string const &path, uint32_t flags, uint32_t mode);
+  ~File();
+
+public:
+  File(const File &other) = delete;
+  File &operator=(const File &other) = delete;
+
+public:
+  File(File &&other) { *this = std::move(other); }
+
+  File &operator=(File &&other) {
```

Comment:
Actually, I don't think you can do that: http://stackoverflow.com/questions/32796739/can-i-call-destructor-in-move-assignment-operator